src/agnocastlib/test/integration/test_agnocast_heaphook.cpp

```diff
@@ -1,7 +1,102 @@
+#include "agnocast/agnocast_publisher.hpp"
+#include "agnocast/agnocast_utils.hpp"
+
+#include <rclcpp/rclcpp.hpp>
+
 #include <gtest/gtest.h>
 
-TEST(agnocast_heaphook, dummy)
+#include <cstdint>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+extern "C" {
+uint32_t agnocast_get_borrowed_publisher_num();
+}
+
+namespace agnocast
+{
+void increment_borrowed_publisher_num();
+void decrement_borrowed_publisher_num();
+}  // namespace agnocast
+
+bool is_address_in_shared_mapping(void * address)
 {
-  // Dummy test to avoid "No tests found in test_agnocast_heaphook.cpp" error
-  SUCCEED();
+  if (!address) {
+    return false;
+  }
+
+  std::ifstream maps_file("/proc/self/maps");
+  if (!maps_file.is_open()) {
+    return false;
+  }
+
+  std::string line;
+  uintptr_t addr = reinterpret_cast<uintptr_t>(address);
+
+  while (std::getline(maps_file, line)) {
+    std::istringstream iss(line);
+    std::string addr_range, perms, offset, dev, inode;
+
+    if (!(iss >> addr_range >> perms >> offset >> dev >> inode)) {
+      continue;
+    }
```

Comment:
Could you simply remove `dev` and `inode`?
They are unnecessary for address range check.